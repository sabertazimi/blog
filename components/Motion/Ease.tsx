import type { MotionProps } from '@components/utils'
import { motion } from '@components/utils'
import type { ReactNode } from 'react'

interface Props {
  children: ReactNode
  className?: string
  delay?: number
  duration?: number
}

const initialVariants: MotionProps['initial'] = {
  opacity: 0,
}

const animateVariants: MotionProps['animate'] = {
  opacity: 1,
}

const viewport: MotionProps['viewport'] = {
  once: true,
}

const Ease = ({
  children,
  className,
  delay = 0,
  duration = 1,
  ...props
}: Props): JSX.Element => (
  <motion.div
    className={className}
    initial={initialVariants}
    animate={animateVariants}
    transition={{
      delay,
      duration,
    }}
    viewport={viewport}
    {...props}
  >
    {children}
  </motion.div>
)

export default Ease
