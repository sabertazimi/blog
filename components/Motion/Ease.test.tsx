import { render } from '@utils'
import Ease from './Ease'

describe('ease', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(
      <Ease>
        <div>Ease</div>
      </Ease>,
    )

    expect(container).toMatchSnapshot()
  })
})
