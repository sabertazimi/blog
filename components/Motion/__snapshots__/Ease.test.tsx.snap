// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ease should render correctly (snapshot) 1`] = `
<div>
  <div
    style="opacity: 0;"
  >
    <div>
      Ease
    </div>
  </div>
</div>
`;
